File: typescript/src/issue_action.ts
import { createNodeClient } from "./client"
import {create} from "@bufbuild/protobuf";
import {IssueActionRequestSchema} from "@buf/nativeconnect_api.bufbuild_es/nativeconnect/api/v1/nativeconnect_pb";

const client = createNodeClient()

const request = create(IssueActionRequestSchema, {
  vin: "",
  command: {
    case: "lock",
    value: {
      locked: true
    }
  }
})

// Returns empty
await client.issueAction(request)

File: python/issue_action.py
from client import create_native_client
import compassiot.native.v1.native_pb2 as native


def main():
    client = create_native_client()

    request = native.IssueActionRequest(
        vin="",
        lock=native.SetLockCommand(
            locked=True
        )
    )

    # Returns empty, check using NativeGetLatestPoint
    client.NativeIssueAction(request)

     
if __name__ == "__main__":
    main()
